src/libcephfs_proxy/proxy_mount.c

```diff
@@ -470,7 +470,7 @@ int32_t proxy_path_resolve(proxy_mount_t *mount, const char *path,
 	}
 
 	if (err >= 0) {
-		err = proxy_path_lookup(proxy_cmount(mount), iter.base, ".",
+		err = proxy_path_lookup(proxy_cmount(mount), iter.base, "",
```

Comment:
lookups should not be allowed using a non-dir parent inode. It should return ENOTDIR as per POSIX compliance and that's what the patch (soon pushed in https://github.com/ceph/ceph/pull/62488 after I resolve the mysterious cap leak caused by path walking `.` using a file node) would do, so I guess this is a temp workaround for the SMB clients to not get blocked, right? because once the patch is out and if the `iter.base` is a file inode then it would return ENOTDIR. So maybe you'd need to align this code again.

Reply:
I'm not sure posix defines a strict interface for low-level access to inodes. But I agree that it doesn't seem to make much sense.

However, there's no other way to get a new reference to an inode using libcephfs. Without that, we require this.

Reply:
however im not sure how much feasible that would be. just thinking out loud.

Reply:
Oh..I see that we already have an internal implementation _ll_get()_. May be it is as simple as creating and exposing it as _ceph_ll_get()_?

Reply:
`_ll_get()` is used to get a ll_ref on the inode, also the ret type is void.

Reply:
I think we can have something like `AT_FDCWD` case https://sites.uclouvain.be/SystInfo/usr/include/linux/fcntl.h.html as well.

Reply:
> `_ll_get()` is used to get a ll_ref on the inode, also the ret type is void.

That is all what is required if we move forward with disallowing lookup with non-directory parent inode, right? Please correct me if I am wrong.

Reply:
> > `_ll_get()` is used to get a ll_ref on the inode, also the ret type is void.
> 
> That is all what is required if we move forward with disallowing lookup with non-directory parent inode, right? Please correct me if I am wrong.

Yes. That would be enough. Those references should be regular ones that can be released with ceph_ll_put(). With that, the "special" lookup is not required anymore.

Reply:
> Oh..I see that we already have an internal implementation __ll_get()_. May be it is as simple as creating and exposing it as _ceph_ll_get()_?

I'll try this approach and see if it can satisfy our requirements.